### DISCUSSION TITLE
page level data binding/reference

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3085] [Ideas]
Hello,

I am new to budibase, and really love the budibase concept, UI design, and especially the performance, it's really fast. But there's one thing I couldn't quite understand, why budibase have a nesting data binding restrictions, it's really something less flexible. Before move to budibase, I have tried Retool and appsmith, both have page level binding/access, and easy to access to table/repeater or other widget/components property anyway in the same page. Although there's some workaround, but it will be perfect If budibase will have more flexibility later. Anyway, after tried many low-code/no-code platform, budibase is the most responsive, stable and have the most beautiful UI. Hope budibase will have this kind of improvement later.

**Comment by mjashanks** [comment]
Thanks so much :)

I agree - the nesting can be annoying. I'm not 100% sure why this is, but we'll look into it.
